docs/src/codegen-intro.md

```diff
@@ -28,18 +35,82 @@ playwright codegen demo.playwright.dev/todomvc
 pwsh bin/Debug/netX/playwright.ps1 codegen demo.playwright.dev/todomvc
 ```
 
-Run `codegen` and perform actions in the browser. Playwright will generate the code for the user interactions. `Codegen` will attempt to generate resilient text-based selectors.
+### Recording a test
+* langs: js
 
-<video width="100%" height="100%" controls muted >
-  <source src="https://user-images.githubusercontent.com/13063165/197979804-c4fa3347-8fab-4526-a728-c1b2fbd079b4.mp4" type="video/mp4" />
-  Your browser does not support the video tag.
-</video>
+Run `codegen` and perform actions in the browser. Playwright will generate the code for the user interactions. `Codegen` will look at your page and figure out the best locator, prioritizing role, text and test id locators. If the generator finds multiple elements matching the locator, it will improve the locator to make it resilient and uniquely identify the target element, so you don't have to worry about failing tests due to locators.
```

Comment:
Worth changing `best locator` to 'recommended locator'. also some minor sentence suggestions.

Also: `so you don't have to worry about failing tests due to locators`  could change to 'therefore eliminating test/s failure and flakiness due to locators

```suggestion
Run `codegen` and perform actions in the browser. Playwright will generate the code for the user interactions. `Codegen` will look at the rendered page and figure out the recommended locator, prioritizing role, text and test id locators. If the generator identifies multiple elements matching the locator, it will improve the locator to make it resilient and uniquely identify the target element, therefore eliminating and reducing test/s failing and flaking due to locators
```